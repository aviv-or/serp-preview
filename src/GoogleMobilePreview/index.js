import React from 'react'
import styled from 'styled-components'
import {default as PT} from 'prop-types'

const Wrapper = styled.div`
  background-color: white;
  line-height: 20px;
  -webkit-box-shadow: rgba(32,33,36,0.28) 0px 1px 6px;
  box-shadow: rgba(32,33,36,0.28) 0px 1px 6px;
  border-radius: 8px;
  max-width: 342px;
  width: auto;
  padding: 12px 16px;
  margin-bottom: 10px;
  color: #202124;
  font-family: arial, sans-serif;
`
const Title = styled.a`
  font-size: 16px;
  line-height: 20px;
  color: #1967D2;
  height: auto;
  min-height: 1em;
  text-decoration: none;
  &:hover {
    text-decoration: underline;
  }
`
const Description = styled.div`
  font-size: 14px;
  color: #3C4043;
  line-height: 1.58;
  margin-top: 7px;
`
const Url = styled.div`
  height: 19px;
  font-size: 12px;
  margin: 0 0 9px 0;
  cursor: pointer;
  color: #5f6368;
  max-width: 100%;
  padding-bottom: 1px;
  padding-top: 1px;
  display: flex;
  align-items: center;
`
const Icon = styled.img`
  width: 14px;
  height: 14px;
  margin-right: 8px;
`
const DomainWrapper = styled.span`
  white-space: nowrap;
  overflow: hidden;
  -o-text-overflow: ellipsis;
  text-overflow: ellipsis;
`
const DomainText = styled.span`
  color: #202124;
  font-size: 12px;
`

const formatUrl = (url) => url
  .replace(/https?:\/\//, '')
  .replace(/\//g, ' â€º ')

const GoogleMobilePreview = ({ title, description, url, link }) => {
  const domain = url ? formatUrl(url) : ''
  return (
    <Wrapper>
      <Url>
        <Icon src={`https://www.google.com/s2/favicons?domain=${domain}`} />
        <DomainWrapper>
          <DomainText>{domain}</DomainText>
        </DomainWrapper>
      </Url>
      <Title target='__blank' href={link}>{title}</Title>
      <Description>{description}</Description>
    </Wrapper>
  )
}

GoogleMobilePreview.defaultProps = {
  description: null,
  link: '#'
}

GoogleMobilePreview.propTypes = {
  /** Title */
  title: PT.string.isRequired,
  /** Meta Description */
  description: PT.oneOfType([
    PT.string,
    PT.arrayOf(PT.node),
    PT.node
  ]),
  /** Url */
  url: PT.string.isRequired,
  /** Link for src */
  link: PT.string
}

export default GoogleMobilePreview
