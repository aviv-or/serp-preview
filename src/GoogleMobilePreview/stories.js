import React from 'react'
import { storiesOf } from '@storybook/react'
import GoogleMobilePreview from './'

const testResults = [
  {
    title: 'ðŸ¥‡React â€“ A JavaScript library for building user interfaces',
    description: 'React makes it painless to create interactive UIs. Design simple views for each state in your application, and React will efficiently update and render just the right components when your data changes. Declarative views make your code more predictable and easier to debug.',
    url: 'https://reactjs.org'
  },
  {
    title: 'facebook/react: A declarative, efficient, and flexible ... - GitHub',
    description: <>React. Â· <em>React</em> is a JavaScript library for building user interfaces. Declarative: React makes it painless to create interactive UIs. Design simple views for each state in your application, and React will efficiently update and render just the right components when your data changes.</>,
    url: 'https://github.com/facebook/react'
  },
  {
    title: 'React Native Â· A framework for building native apps using React',
    description: 'A framework for building native apps using React.',
    url: 'https://reactnative.dev'
  }
]

storiesOf('Google Mobile')
  .add('Default', () => (
    testResults.map(({ title, description, url }, idx) =>
      <figure style={{marginBottom: '27px'}} key={idx}>
        <GoogleMobilePreview title={title} description={description} url={url} />
      </figure>
    )
  ))
