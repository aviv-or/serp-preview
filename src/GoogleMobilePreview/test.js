import React from 'react'
import GoogleMobilePreview from '.'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

it('show preview', () => {
  const tree = renderer
    .create(
      <GoogleMobilePreview
        title="test title"
        metadescription="meta description"
        url="test.com" />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
