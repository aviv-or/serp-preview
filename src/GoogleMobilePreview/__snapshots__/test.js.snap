// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`show preview 1`] = `
.c0 {
  background-color: white;
  line-height: 20px;
  -webkit-box-shadow: rgba(32,33,36,0.28) 0px 1px 6px;
  box-shadow: rgba(32,33,36,0.28) 0px 1px 6px;
  border-radius: 8px;
  max-width: 342px;
  width: auto;
  padding: 12px 16px;
  margin-bottom: 10px;
  color: #202124;
  font-family: arial,sans-serif;
}

.c5 {
  font-size: 16px;
  line-height: 20px;
  color: #1967D2;
  height: auto;
  min-height: 1em;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c6 {
  font-size: 14px;
  color: #3C4043;
  line-height: 1.58;
  margin-top: 7px;
}

.c1 {
  height: 19px;
  font-size: 12px;
  margin: 0 0 9px 0;
  cursor: pointer;
  color: #5f6368;
  max-width: 100%;
  padding-bottom: 1px;
  padding-top: 1px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  width: 14px;
  height: 14px;
  margin-right: 8px;
}

.c3 {
  white-space: nowrap;
  overflow: hidden;
  -o-text-overflow: ellipsis;
  text-overflow: ellipsis;
}

.c4 {
  color: #202124;
  font-size: 12px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <img
      className="c2"
      src="https://www.google.com/s2/favicons?domain=test.com"
    />
    <span
      className="c3"
    >
      <span
        className="c4"
      >
        test.com
      </span>
    </span>
  </div>
  <a
    className="c5"
    href="#"
    target="__blank"
  >
    test title
  </a>
  <div
    className="c6"
  />
</div>
`;
